### PR TITLE
Code format using clang-format

### PR DESCRIPTION
**Pull request by janLo**
This runs clang-format over all source files to have a consistent and reproducible formatting.
A note how to use it was added to README.md.

### CODE REVIEW

**Comment by janLo** [comment]
This is part of the effort to split #14.

File: src/ESPiLight.cpp [hunk 1]
```diff
@@ -78,60 +79,55 @@ void ICACHE_RAM_ATTR ESPiLight::interruptHandler() {
...
 
-void ESPiLight::enableReceiver() {
-  _enabledReceiver = true;
-}
+void ESPiLight::enableReceiver() { _enabledReceiver = true; }
 
-void ESPiLight::disableReceiver() {
-  _enabledReceiver = false;
-}
+void ESPiLight::disableReceiver() { _enabledReceiver = false; }
```
Comment (puuu) [inline src/ESPiLight.cpp]: Can we avoid this one-line style?

Comment (janLo) [reply src/ESPiLight.cpp]: You can do a lot, its just fiddly to divert from the default styles: https://clang.llvm.org/docs/ClangFormatStyleOptions.html

I think what you want in particular is:
```
AllowShortFunctionsOnASingleLine: None
```
or
```
AllowShortFunctionsOnASingleLine: Empty
```

Comment (puuu) [reply src/ESPiLight.cpp]: No, thanks. It is not worth for a project like this.

**Comment by puuu** [comment]
merged in 322ca6d028fd9df69f6c3ead552c00c17771d121

**Pull request closed without merging**